broker/src/main/java/org/apache/rocketmq/broker/processor/PopReviveService.java

```diff
@@ -225,7 +224,11 @@ public PullResult getMessage(String group, String topic, int queueId, long offse
                 getMessageResult.getMaxOffset(), foundList);
 
         } else {
-            POP_LOGGER.error("get message from store return null. topic={}, groupId={}, requestOffset={}", topic, group, offset);
+            long maxQueueOffset = brokerController.getMessageStore().getMaxOffsetInQueue(topic, queueId);
```

Comment:
Log always if a null result is returned. I think this is an exceptional case. For the typical case, a clear result should be returned, even if `offset > maxOffset`(OFFSET_OVERFLOW_BADLY is returned).

Reply:
At present, a clean start-up would observe bulks of errors in the server. This is unwanted and intimidating to new users. IMO, it's good to log conditionally and add potential ones if there is a need.